ACKImagePicker/ViewControllers/ImagePickerViewController.swift

```diff
@@ -350,8 +362,38 @@ extension ImagePickerViewController: ACKImagePickerDelegate {
         
         // call `onImagesPicked` block on main thread when whole group is finished
         group.notify(queue: .main) { [weak self] in
-            currentViewController?.stopLoadingAnimation()
+            progressView?.removeFromSuperview()
+            overlayView?.removeFromSuperview()
+            currentViewController?.view.isUserInteractionEnabled = true
+            self?.stopLoadingAnimation()
             self?.onImagesPicked?(images)
         }
     }
+    
+    private func setupUI(in currentViewController: BaseViewController?) -> (progressView: UIProgressView?, overlayView: UIView?) {
+        guard let currentViewController = currentViewController else { return (progressView: nil, overlayView: nil) }
+        let overlayView = UIView()
+        overlayView.backgroundColor = UIColor.black.withAlphaComponent(0.5)
+        overlayView.isUserInteractionEnabled = false
+        currentViewController.view.addSubview(overlayView)
+        overlayView.translatesAutoresizingMaskIntoConstraints = false
+        overlayView.leadingAnchor.constraint(equalTo: currentViewController.view.leadingAnchor).isActive = true
+        overlayView.trailingAnchor.constraint(equalTo: currentViewController.view.trailingAnchor).isActive = true
+        overlayView.bottomAnchor.constraint(equalTo: currentViewController.view.bottomAnchor).isActive = true
+        overlayView.topAnchor.constraint(equalTo: currentViewController.view.topAnchor).isActive = true
+        
+        let progressView = UIProgressView()
+        currentViewController.view.addSubview(progressView)
+        progressView.translatesAutoresizingMaskIntoConstraints = false
+        progressView.heightAnchor.constraint(equalToConstant: 5).isActive = true
+        progressView.leadingAnchor.constraint(equalTo: currentViewController.view.leadingAnchor, constant: 0).isActive = true
+        progressView.trailingAnchor.constraint(equalTo: currentViewController.view.trailingAnchor, constant: 0).isActive = true
```

Comment:
Why not use a more efficient `NSLayoutConstraint.activate(_:)` that does the layout as a single batch, instead of activating the constraints one by one?

Reply:
Yeah, good catch, fixed 👍